File: aricode/sparring/challenges/02_fib/fib.c
/* C - Challenge 02: Fibonacci */
/* Compute fib(10) = 55, return as exit code */

int fib(int n) {
    if (n < 2) return n;
    return fib(n - 1) + fib(n - 2);
}

int main(void) {
    return fib(10);
}

File: aricode/sparring/challenges/02_fib/fib.go
// Go - Challenge 02: Fibonacci
// Compute fib(10) = 55, return as exit code

package main

import "os"

func fib(n int) int {
	if n < 2 {
		return n
	}
	return fib(n-1) + fib(n-2)
}

func main() {
	os.Exit(fib(10))
}
